pkg/controller/jenkins/reconciler_test.go

```diff
@@ -0,0 +1,181 @@
+package jenkins
+
+import (
+	"testing"
+	"time"
+
+	"reflect"
+
+	"github.com/redhat-developer/openshift-jenkins-operator/test/mocks"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+	resource "k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+const (
+	test_ns   = "test"
+	test_name = "test-jenkins"
+)
+
+// func TestReconcilerCreateResourceIfNotPresent(t *testing.T) {
+// 	t.FailNow()
+// }
+
+func TestNewJenkinsService(t *testing.T) {
+	t.Run("TestNewJenkinsService", func(t *testing.T) {
+		jenkinsPort := corev1.ServicePort{
+			Name:     JenkinsWebPortName,
+			Port:     JenkinsWebPort,
+			Protocol: JenkinsWebPortProtocol,
+			TargetPort: intstr.IntOrString{
+				IntVal: JenkinsWebPortAsInt,
+				StrVal: JenkinsWebPortAsStr,
+			},
+		}
+
+		checkSvc := corev1.Service{
+			TypeMeta: metav1.TypeMeta{Kind: "", APIVersion: ""},
```

Comment:
By the way, most probably you don't need a `TypeMeta`, please consider: https://github.com/operator-framework/operator-sdk/blob/e0342c4f1c34e961520956baaadb5851cc11a8ac/doc/user/unit-testing.md#testing-with-3rd-party-resources